### PR TITLE
Bug 1862896: add requirements section and generate-in-docker target

### PR DESCRIPTION
**Pull request by paulfantom**
Putting requirements section into CONTRIBUTING.md to show which packages are needed to be installed before doing a contribution.

Re-adding `make generate-in-docker` as a workaround solution until we remove python dependency.

/cc @s-urbaniak @lilic 

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,6 +30,18 @@ These steps outline the general contribution workflow:
 * Make sure the tests pass, and add any new tests as appropriate.
 * Submit a pull request to the original repository. (see [Format of Pull Requests](#format-of-pull-requests))
 
+## Required tools
+To allow scripts and `make` targets working correctly, ensure you have following tools installed in your system:
+
+* golang
```
Comment (lilic) [inline CONTRIBUTING.md:36]: 1.14+ I believe

Nit:
```suggestion
* go v1.14+
```

Comment (paulfantom) [reply CONTRIBUTING.md:36]: This is a moving target hence I didn't put version number in it. We require specific version number in `go.mod`

Comment (lilic) [reply CONTRIBUTING.md:36]: We can always update if need specific versions later, but it won't work pre 1.14 no?

Comment (paulfantom) [reply CONTRIBUTING.md:36]: This is exactly what locking in `go.mod` is for. No need to add documentation entry which can easily go stale.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,6 +30,18 @@ These steps outline the general contribution workflow:
...
 
+## Required tools
+To allow scripts and `make` targets working correctly, ensure you have following tools installed in your system:
+
+* golang (see `go.mod` file for minimum required go version)
+* `awk` (GNU variant)
+* `sed` (GNU variant)
+* `make`
+* `curl`
+* python and pyyaml package
```
Comment (elmiko) [inline CONTRIBUTING.md:41]: you might want to call out that the tooling is currently looking for python2, this caused some surprise for me when trying to install pyyaml.

Comment (lilic) [reply CONTRIBUTING.md:41]: good point! @paulfantom

Comment (paulfantom) [reply CONTRIBUTING.md]: Added.

**Comment by paulfantom** [comment]
I added changes from #897 to prevent merge conflicts and possible rebases. Both PRs were trying to solve parts of the same BZ and both would need to land in order to fully satisfy BZ requirements.

/unhold

**Comment by s-urbaniak** [comment]
uhh, one of those flakes again, investigating.

**Pull request merged**